src/app.rs

```diff
@@ -1165,7 +1186,14 @@ impl SearchState {
     }
 
     fn is_string_match(&self, s: &str) -> bool {
-        s.contains(&self.query)
+        if self.whole_word {
+            match &self.last_word_regex {
+                Some(regex) => regex.is_match(s),
+                _ => false,
```

Comment:
This is a matter of taste, but I would write this:

```rust
let Some(regex) = &self.last_word_regex else {
  unreachable!();
}
regex.is_match(s)
```

There are a couple reasons to do this. First, we don't want to just ignore an error here. So the `unreachable!()` signals this is a branch we shouldn't ever hit (and is an error if we do). Because we know the else branch is unreachable, we can optimize the happy path with `let ... else`, meaning that the variable `regex` is always defined subsequent to this statement, and reducing the indentation of the continuation.

Reply:
For posterity, the other way to write this is:

```rust
let regex = self.last_word_regex.unwrap();
regex.is_match(s)
```

That's fine too if you want to write it that way. It's shorter, less explicit, and it's easier to miss the `.unwrap()` (which panics if the option is not `Some(...)`). I don't have a strong preference, but I feel the profiler backend suffers a bit from using `unwrap()` all over the place.

Reply:
[ffda863](https://github.com/StanfordLegion/prof-viewer/pull/32/commits/ffda863c1685f9b6af8fc2b3d88f2b092bcce10c)

Definitely better, still acclimating to `let Some(...)` and matching in general, honestly reminds me (from the mists of memory) of Prolog.